codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketChunkedInput.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.websocketx;
+
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.websocketx.ContinuationWebSocketFrame;
+import io.netty.handler.codec.http.websocketx.WebSocketFrame;
+import io.netty.handler.stream.ChunkedInput;
+
+/**
+ * A {@link ChunkedInput} that fetches data chunk by chunk for use with WebSocket chunked transfers.
+ * <p>
+ * Each chunk from the input data will be wrapped within a {@link WebSocketFrame}. At the end of the input data,
+ * {@link WebSocketFrame} with finalFragment will be written.
+ * <p>
+ */
+public class WebSocketChunkedInput implements ChunkedInput<WebSocketFrame> {
```

Comment:
@zizigooloo shouldnt the first frame be a `BinaryWebSocketFrame` or a `TextWebSocketFrame` ?